gwu-libraries/sfm-twitter-harvester
# T1118 tests for exporter2

Resolves gwu-libraries/sfm-ui#1118

### Contains
- @sebastian-nagel's fix to the unit tests for the user timeline harvest (Twitter v. 2 API)
- Tests for the Twitter rest exporter and WARC iter code (Twitter v. 2 API), using both CSV and full JSON formats
- Tests for exporting selected seeds from a user timeline v. 2 harvest
- Integration tests for Twitter v.2 search, user timeline, and filtered stream harvests

### Testing
- Verify that all unit tests pass: `python -m unittest -v`
- Create `tests/test_config.py` file to hold your Twitter credentials for integration testing:
```
TWITTER_CONSUMER_KEY = ''
TWITTER_CONSUMER_SECRET = ''
TWITTER_ACCESS_TOKEN = ''
TWITTER_ACCESS_TOKEN_SECRET = ''
TWITTER_BEARER_TOKEN = ''
```
- Make sure the `docker/ci.docker-compose.yml` is building from local images for the `twitterrestharvester` and `twitterstreamharvester` service, e.g.:
```
      build:
          context: ..
          dockerfile: Dockerfile-rest-harvester
```
- Bring up the testing containers: `docker-compose -f docker/ci.docker-compose.yml up -d`
- Run tests and verify that they pass: `docker exec docker_sfmtwitterstreamharvester_1 python -m unittest -v` and `docker exec docker_sfmtwitterrestharvester_1 python -m unittest -v`
- Clean up:
```
docker-compose -f docker/ci.docker-compose.yml kill
docker-compose -f docker/ci.docker-compose.yml rm -v --force
```
